VM_SETUP.md

```diff
@@ -0,0 +1,179 @@
+  ipv4.dhcp: "false"
+  ipv4.nat: "true"
+  ipv6.address: none
+description: ""
+name: maas-kvm
+type: bridge
+used_by: []
+managed: true
+status: Created
+locations:
+- none
+__EOF
+
+lxc config set core.https_address [::]:8443
+```
+
+## LXD Profile Setup
+
+Next, we need to create an LXD profile for our VMs. This will set memory and cpu requirements, configure the default installed packages, copy your ssh key into the VMs so you can access them, and mount your clone of `maas-anvil` into the VM.
+
+```bash
+# This needs to point to where you have maas-anvil cloned
+MAAS_ANVIL_SRC="/home/your_username/maas-repos/maas-anvil"
+PROFILE_NAME="anvil"
+
+lxc profile create ${PROFILE_NAME}
+cat <<EOF | lxc profile edit ${PROFILE_NAME}
+config:
+    limits.memory: 4GB
+    limits.cpu: "4"
+    raw.idmap: |
+        uid $(id -u) 1000
+        gid $(id -g) 1000
+    user.vendor-data: |
+        #cloud-config
+        packages:
+        - git
+        - build-essential
+        - jq
+        - snapcraft
+        runcmd:
+        - cat /dev/zero | ssh-keygen -q -N ""
+        ssh_authorized_keys:
+        - $(cat ${HOME}/.ssh/id_rsa.pub | cut -d' ' -f1-2)
+description: MAAS Anvil environment
+devices:
+    root:
+        path: /
+        pool: default
+        type: disk
+        size: 50GB
+    work:
+        type: disk
+        source: ${MAAS_ANVIL_SRC}
+        path: /work
+    eth0:
+        type: nic
+        name: eth0
+        network: maas-anvil-ctrl
+    eth1:
+        type: nic
+        name: eth1
+        network: maas-anvil-kvm
+EOF
+```
+
+## Launch the VMs
+
+To create your VMs, execute the following:
+
+```bash
+lxc launch ubuntu:22.04 infra1 --vm -p ${PROFILE_NAME}
+lxc launch ubuntu:22.04 infra2 --vm -p ${PROFILE_NAME}
+lxc launch ubuntu:22.04 infra3 --vm -p ${PROFILE_NAME}
+```
+
+If you are running these commands in a script, include the lines below to wait for the VMs to be ready.
+
+```bash
+# this sleep may not be necessary, or may be shortened depending on your system
+sleep 10
+lxc exec infra1 -- cloud-init status --wait
+lxc exec infra2 -- cloud-init status --wait
+lxc exec infra3 -- cloud-init status --wait
+```
+
+## Set up VMs Internally and Install MAAS Anvil
+
+We now need to execute some commands in each VM. This will set up our second network interface, setup LXD inside the VM, and build & install the MAAS Anvil snap.
+
+```bash
+KVM_NETWORK_PREFIX="10.40.0"
+
+sudo tee /etc/netplan/99-maas-kvm-net.yaml <<EOF
+network:
+    version: 2
+    ethernets:
+        enp6s0:
+            addresses:
+                - ${KVM_NETWORK_PREFIX}.2/24
+EOF
+sudo netplan apply
+
+lxd init --auto
+
+sleep 5
+cd /work
+/usr/bin/make snap
+sudo snap install ./maas-anvil.snap --dangerous
+```
+
+To execute this inside the VM from a script running on your host machine, save the contents above to a file (below, `/path/to/script`), then run:
+
+```bash
+infra1_ip=$(lxc list -c4 --format csv infra1 | grep -o "10.30.0.[0-9][0-9]")
```

Comment:
I've found this method can be a little flaky if a container has multiple IPs, I've been using
`infra1_ip=$(lxc exec infra1 -- hostname -I | cut -d " " -f 1)`
instead for a few months, which seems much more stable